anyway/flask_app.py

```diff
@@ -1500,15 +1500,14 @@ def add_to_role() -> Response:
     return change_user_roles("add")
 
 
-def check_admin_api_with_input(
+def look_for_errors_in_request_admin(
     request: Request, allowed_fields: typing.List[str]
 ) -> typing.Optional[Response]:
     if current_user.is_anonymous:
         return return_json_error(Es.BR_USER_NOT_LOGGED_IN)
```

Comment:
If you map current user to a role, and then decide the permission according to the role, the check for anonymous can be avoided.